addons/http_routing/models/ir_http.py

```diff
@@ -223,8 +239,8 @@ def _get_converters(cls):
         )
 
     @classmethod
-    def _get_language_codes(cls):
-        return request.env['res.lang'].get_installed()
+    def _get_languages(cls):
+        return request.env['res.lang'].search([])
```

Comment:
if you don't want to use the `get_installed` method, maybe at least make your own ormcached method (`get_language_ids` or something) and be sure to have the right `active_test` context (as does `get_installed`)

Reply:
Before [85ee046b3](https://github.com/odoo-dev/odoo/commit/85ee046b37c7c9a0dc7f052b8258f81774b127de) the code was using the `_get_language_codes()` method declared in `http_routing` so it was easy to change.

So it's not that I don't want to use `get_installed`, but it is returning a tupple in `base`, with only (code, name) while I'd need `url_code` too from `http_routing`, so after rebasing on top of your commit, it was easier to replace it by a `search([])` like it was before in `_get_language_codes()` to move forward on what I was working on :-)..

As you can see by the list of commits in this PR, it is not finished yet.

There is also a 'to triple check/confirm' commit which is removing some old cached method. Did not yet fully investigated it.

But appreciate the advice, once dev is done you might wanna have a last look at it.